src/Cassandra/Metadata.cs

```diff
@@ -133,14 +148,120 @@ public IEnumerable<IPEndPoint> AllReplicas()
             return Hosts.AllEndPointsToCollection();
         }
 
-        internal void RebuildTokenMap()
+        internal async Task<bool> RebuildTokenMapAsync(bool retry)
+        {
+            var currentTicks = DateTime.UtcNow.Ticks;
+            
+            // running this statement synchronously inside the exclusive scheduler deadlocks
+            var ksList = await _schemaParser.GetKeyspaces(retry).ConfigureAwait(false);
+
+            var task = _tokenMapTaskFactory.StartNew(() =>
+            {
+                var lastRebuildTicks = Interlocked.Read(ref _lastRebuildTicks);
+                if (lastRebuildTicks > currentTicks)
+                {
+                    return true;
+                }
+                Interlocked.Exchange(ref _lastRebuildTicks, currentTicks);
+                
+                Metadata.Logger.Info("Retrieving keyspaces metadata");
```

Comment:
We could change this one to `"Updating keyspaces metadata"` and include one above schema parser call that logs the retrieval.

Reply:
sounds good, done